src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -270,6 +274,24 @@ export default class AdyenV2PaymentStrategy implements PaymentStrategy {
         });
     }
 
+    private _handleFromAction(action: AdyenV2Action, paymentMethodId: string): Promise<any> {
+        return new Promise(() => {
+            if (!this._adyenCheckout) {
+                throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+            }
+
+            if (paymentMethodId === AdyenV2PaymentMethodType.IDEAL) {
+                const { onLoad } = this._getAdyenV2PaymentInitializeOptions().threeDS2Options;
```

Comment:
Since you've decided to generalise 3DS and other additional payment actions, you'll need to deprecate this `threeDS2Options` field as well and introduce a new field, something like `additionalActionOptions`. Similarly, you'll need to read from both in places where `threeDS2Options` field is used at the moment.